packages/dapi/test/unit/grpcServer/handlers/tx-filter-stream/subscribeToTransactionsWithProofsHandlerFactory.spec.js

```diff
@@ -321,61 +268,111 @@ describe('subscribeToTransactionsWithProofsHandlerFactory', () => {
       expect(call.end).to.not.have.been.called();
     }
   });
 
   it('should respond with Not Found error if fromBlockHash is not part of the best block chain', async () => {
-    const blockHash = 'someBlockHash';
-    call.request.setFromBlockHash(blockHash);
+    const blockHash = '00000bafbc94add76cb75e2ec92894837288a481e5c005f6563d91623bf8bc2c';
+    call.request.setFromBlockHash(Buffer.from(blockHash, 'hex'));
     call.request.setCount(0);
 
     const error = new Error();
-    error.code = -5;
+    error.code = -8;
 
-    coreAPIMock.getBlock.resolves({ height: 1, confirmations: -1 });
+    coreAPIMock.getBlockStats.throws(error);
 
     try {
       await subscribeToTransactionsWithProofsHandler(call);
 
       expect.fail('should throw NotFoundGrpcError');
     } catch (e) {
       expect(e).to.be.instanceOf(NotFoundGrpcError);
-      expect(e.getMessage()).to.equal(
-        `block ${Buffer.from(blockHash).toString('hex')} is not part of the best block chain`,
-      );
+      expect(e.getMessage()).to.equal(`Block ${blockHash} not found`);
 
       expect(call.write).to.not.have.been.called();
       expect(call.end).to.not.have.been.called();
     }
   });
+
+  it('should respond with error if `fromBlockHeight is 0 and `fromBlockHash` is not set', async () => {
+    const bloomFilterMessage = new BloomFilter();
+
+    bloomFilterMessage.setVData(new Uint8Array());
+    bloomFilterMessage.setNTweak(1000);
+    bloomFilterMessage.setNFlags(100);
+    bloomFilterMessage.setNHashFuncs(10);
+
+    const request = new TransactionsWithProofsRequest();
+
+    request.setFromBlockHeight(0);
+    request.setBloomFilter(bloomFilterMessage);
+
+    call.request = request;
+
+    try {
+      await subscribeToTransactionsWithProofsHandler(call);
+
+      expect.fail('should fail with InvalidArgumentGrpcError');
+    } catch (e) {
+      expect(e).to.be.an.instanceOf(InvalidArgumentGrpcError);
+      expect(e.getMessage()).to.equal('Minimum value for `fromBlockHeight` is 1');
+      expect(call.write).to.not.have.been.called();
+      expect(call.end).to.not.have.been.called();
+      expect(getMemPoolTransactionsMock).to.not.have.been.called();
+    }
+  });
+
+  it('should respond with error if if both fromBlockHash and fromBlockHeight are not specified', async () => {
+    const bloomFilterMessage = new BloomFilter();
+
+    bloomFilterMessage.setVData(new Uint8Array());
+    bloomFilterMessage.setNTweak(1000);
+    bloomFilterMessage.setNFlags(100);
+    bloomFilterMessage.setNHashFuncs(10);
+
+    const request = new TransactionsWithProofsRequest();
+
+    request.setBloomFilter(bloomFilterMessage);
+
+    call.request = request;
+
+    try {
+      await subscribeToTransactionsWithProofsHandler(call);
+
+      expect.fail('should fail with InvalidArgumentGrpcError');
+    } catch (e) {
+      expect(e).to.be.an.instanceOf(InvalidArgumentGrpcError);
+      expect(e.getMessage()).to.equal('Minimum value for `fromBlockHeight` is 1');
```

Comment:
I guess it must be another error, isn't it?

Reply:
it's actually the same test case as above, because not setting fromBlockHeight results in `0` value